### PR TITLE
Automatic Binning

### PR DESCRIPTION
**Pull request by MuellerSeb**
Closes: #55 #36 

This PR adds a simple automatic binning routine `standard_bins` to the variogram submodule.
It returns an equal division from 0 to `max_dist` with `bin_no` as number of bins. The following rules are applied:
- if `max_dist` is not given, it is set to one third of the box diameter
- if `bin_no` is not given, it is determined by sturges' rule from the number of point combinations

You can now call `gs.vario_estimate` without passing `bin_edges`. In this case, the bin_edges will be determined as described above.

### CODE REVIEW

**Comment by MuellerSeb** [comment]
@LSchueler ready for review. :blush:

File: examples/03_variogram/06_auto_bin_latlon.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ax = krige.plot()
+# plotting lat on y-axis and lon on x-axis
+ax.scatter(pos[1], pos[0], 50, c=field, edgecolors="k", label="input")
+ax.legend()
+
+###############################################################################
+# Looks good, doesn't it?
+#
+# This example shows, that setting up variogram estimation and kriging routines
+# is straight forward with GSTools. ;-)
```
Comment (LSchueler) [inline examples/03_variogram/06_auto_bin_latlon.py:76]: Really nice example!

File: gstools/variogram/binning.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+__all__ = ["standard_bins"]
+
+
+def _sturges(pnt_cnt):
+    return int(np.ceil(2 * np.log2(pnt_cnt) + 1))
+
+
+def _chordal_to_great_circle(dist):
+    return 2 * np.arcsin(np.minimum(np.maximum(np.divide(dist, 2), 0), 1))
```
Comment (LSchueler) [inline gstools/variogram/binning.py:30]: I think this could be put into `gstools/tools`.

**Review by LSchueler — CHANGES_REQUESTED** [review]
This is a really nice convenience addition!

What do you think of moving the `_chordal_to_great_circle` method to a more visible place and making it public?

Appart from that, I think we are ready to merge this PR.

**Comment by LSchueler** [comment]
I fixes some typos, so don't forget to pull.

File: examples/03_variogram/06_auto_bin_latlon.py (hunk 1 above)
Comment (MuellerSeb) [reply examples/03_variogram/06_auto_bin_latlon.py:76]: :blush:

**Comment by MuellerSeb** [comment]
Strange. It was merged, but this PR says it has conflicts and can't be merged. It felt like an internal GitHub Error when I clicked on "merge" and I had to reload the page. Hope nothing was broken.

**Pull request closed without merging**